recipes/cpu_features/all/conanfile.py

```diff
@@ -0,0 +1,62 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class CpuFeaturesConan(ConanFile):
+    name = "cpu_features"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/google/cpu_features"
+    description = "A cross platform C99 library to get cpu features at runtime."
+    topics = ("conan", "cpu", "features", "cpuid")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "fPIC": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True}
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake",
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "_build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        cmake = CMake(self)
+        cmake.definitions["BUILD_PIC"] = self.options.get_safe("fPIC", False)
```

Comment:
It's _not_ there on shared ... reverse logic is confusing 

```suggestion
        cmake.definitions["BUILD_PIC"] = self.options.get_safe("fPIC", True)
```

Reply:
if there is no fPIC option (e.g. `del self.options.fPIC` for Windows), then there is no sense to set fPIC, as it either will have no effect, or just fail (as Visual Studio will not understand `-fPIC` flag).

Reply:
But there's also delete `fPIC` when shared which applies to unix

Reply:
so, for Windows it should be False, and for Linux shared it should be true? that makes things unnecessarily harder

Reply:
No because Conan injects the value and it's _usually_ not manipulated by upstream...

In this case there's custom logic... which overrides the false, https://github.com/google/cpu_features/blob/7ed0b0e50e9b999c0a95c4e15e746585fe0056da/CMakeLists.txt#L32-L35

Upstream does the generic https://github.com/google/cpu_features/blob/a8397ba4591237c17d18e4acc091f5f3ebe7391e/CMakeLists.txt#L145
which CMake knows does nothing.

There's no special logic for windows in the recipe (thankfully for us)

Similar Ex:
 https://github.com/conan-io/conan-center-index/blob/c450d0f1903155da25c3f53431d3444f05489869/recipes/lz4/all/conanfile.py#L55